Story about page

misteu opened the pull request:
- Prepared the about page UI
- included a separate storyboard file + model for the about page
- drawed a simple icon in sketch and included it in the sketch file from #59 
- included dummy json for UI rendering (read from the valid_json for the unit test)

![grafik](https://user-images.githubusercontent.com/35889530/66218981-950dc280-e6ca-11e9-9a68-d7e1eb7b554a.png)

Commit:
added about screen+tabbar icon

Commit:
merged in develop

Commit:
Merge branch 'develop' into StoryAboutPage

Commit:
merged in develop

Commit:
added icon to sketch-file, added dummy json+labels

ppeelen commented on SwiftIsland/About/AboutViewController.swift:
```diff
@@ -0,0 +1,57 @@
+    // TODO: remove and include cacheManager etc.
+    guard let dummyData = TestDataManager().getLocalTestAsset(forPath: "About_valid.json") else { return }
+    about = try? JSONDecoder().decode(About.self, from: dummyData)
+  }
```
As your comment states, it should use the `CacheManager`, not use the `TestDataManager` (which should not be in the SwiftIsland target either)

ppeelen commented on SwiftIsland/About/AboutViewController.swift:
```diff
@@ -0,0 +1,57 @@
+  }
+  
+  @IBAction func openGithubUrl(_ sender: Any) {
+    // e.g. open in Safari
```
I don't think this func is finished, correct?

ppeelen commented on SwiftIsland/About/AboutViewController.swift:
```diff
@@ -0,0 +1,57 @@
+    // e.g. open in Safari
+    print("open safari")
+  }
+  
```
Unneeded whitespaces

ppeelen commented on SwiftIslandTests/Entities/AboutTests.swift:
```diff
@@ -0,0 +1,34 @@
+    XCTAssertEqual(aboutContent?.contributors.count, 5)
+    
+  }
+  
```
Unneeded white spaces

ppeelen commented on SwiftIsland.xcodeproj/project.pbxproj:
```diff
@@ -66,6 +66,13 @@
+		CE84EA2A234783DF00C9C82E /* About_valid.json in Resources */ = {isa = PBXBuildFile; fileRef = CE84EA29234783DF00C9C82E /* About_valid.json */; };
+		CE84EA2C2347843E00C9C82E /* AboutTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = CE84EA2B2347843E00C9C82E /* AboutTests.swift */; };
+		CE84EA2E23478B2E00C9C82E /* About_valid.json in Resources */ = {isa = PBXBuildFile; fileRef = CE84EA29234783DF00C9C82E /* About_valid.json */; };
+		CE84EA2F23478B4000C9C82E /* TestDataManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2ADD086A22ADB01400A90D1E /* TestDataManager.swift */; };
```
This file should not be added to the SwiftIsland target; it is for the test target only.

ppeelen reviewed (changes requested):
Great work! I have some comments and believe some things were not yet finished. I hope you have the possibility to sort those out! :)

misteu committed:
updated PR to requested changes

ppeelen commented:
The API is pushed so the new endpoint you made is now available at http://swiftisland.apptrix.se/about

misteu committed:
added acitivity indicator, tested new endpoint

misteu commented:
ok I implemented the actual request and tested it with the new endpoint. The two funcs (getSchedule and getAbout) are very similar. I added a todo-comment for refactoring.

ppeelen commented on SwiftIsland/Utilities/DataManager.swift:
```diff
@@ -87,6 +88,33 @@ extension DataManager: DataManaging {
     }
   }
+  
+  // Todo: Refactor together with get schedule
```
Would it be a good idea to merge this with `get<T: Listable>(ofType:completion:)`? Resolving this now will lead to not having to create a new issue while resolving this issue.

misteu replied:
so I did refactor it to two functions, one for array types (e.g. [Schedule] or [Area]) and one for non-array types (About). I am not fully satisfied, because I wanted to merge both to one single function, but I had no idea how to handle a generic result type that can be an array type or a non array type.

I still have to learn more about the use of generics. Maybe somebody else can merge both functions ;)

ppeelen replied:
@SpacyRicochet Could you have a look at that? I am unfortunately swamped at the moment.

ppeelen commented on SwiftIsland.xcodeproj/project.pbxproj:
```diff
@@ -66,6 +66,12 @@
 		2AF63107226E5CA300509053 /* Area.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2AF63106226E5CA300509053 /* Area.swift */; };
+		CE84EA252343D63400C9C82E /* About.swift in Sources */ = {isa = PBXBuildFile; fileRef = CE84EA242343D63400C9C82E /* About.swift */; };
+		CE84EA282343DA9A00C9C82E /* AboutViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = CE84EA272343DA9A00C9C82E /* AboutViewController.swift */; };
+		CE84EA2A234783DF00C9C82E /* About_valid.json in Resources */ = {isa = PBXBuildFile; fileRef = CE84EA29234783DF00C9C82E /* About_valid.json */; };
```
This file should not be added to the SwiftIsland target, only the test target.

ppeelen reviewed (changes requested):
Great update, unfortunately I still have some minor thoughts.

SpacyRicochet commented:
Something small I noticed; in the screenshot the tab bar items for 'Map' and 'Mentors' have changed positions relative to the original. Was that on purpose?

misteu commented:
> Something small I noticed; in the screenshot the tab bar items for 'Map' and 'Mentors' have changed positions relative to the original. Was that on purpose?

Ah you are right, I must have dragged it around accidentally 👍 Will fix that, too

misteu committed:
moved tab icons back, refactored DataManager

misteu committed:
merged in develop

misteu committed:
fixed broken Storyboard after merge

funky-monkey commented:
@misteu can this be merged or are there things that need fixing?

misteu committed:
merged in develop, solved merge conflicts

misteu commented:
So I think this can be merged now. I merged in the current develop branch that was updated 2 days ago and solved the new conflicts. 

I did not look for some days into the project and remarked that I also changed some colors to dynamic colors for dark mode. Although that goes beyond the ticket... 🙈 I hope that it is okay :)

nvh commented:
There still seem to be merge conflicts committed to this branch: https://github.com/SwiftIsland/island-app/pull/61/files#diff-b20005bfff36d00426bb74486330cf8fR69

misteu committed:
fixed project-file

misteu committed:
added missing files to project

misteu committed:
updated test

misteu commented:
hey it should work now. I merged develop into that branch and fixed the broken project file.
